core/src/test/java/io/questdb/griffin/DistinctTimeSeriesTest.java

```diff
@@ -0,0 +1,162 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.griffin.engine.functions.rnd.SharedRandom;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.Rnd;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DistinctTimeSeriesTest extends AbstractGriffinTest {
+
+    private static final Log LOG = LogFactory.getLog(DistinctTimeSeriesTest.class);
+
+    @Before
+    public void setUp3() {
+        SharedRandom.RANDOM.set(new Rnd());
+    }
+
+    @Test
+    public void testAllTypes() throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile(
+                    "create table x as (" +
+                            "select" +
+                            " cast(x as int) i," +
+                            " rnd_symbol('msft','ibm', 'googl') sym," +
+                            " round(rnd_double(0)*100, 3) amt," +
+                            " to_timestamp('2018-01', 'yyyy-MM') + x * 720000000 timestamp," +
+                            " rnd_boolean() b," +
+                            " rnd_str('ABC', 'CDE', null, 'XYZ') c," +
+                            " rnd_double(2) d," +
+                            " rnd_float(2) e," +
+                            " rnd_short(10,1024) f," +
+                            " rnd_date(to_date('2015', 'yyyy'), to_date('2016', 'yyyy'), 2) g," +
+                            " rnd_symbol(4,4,4,2) ik," +
+                            " rnd_long() j," +
+                            " timestamp_sequence(500000000000L,x/4) ts," +
+                            " rnd_byte(2,50) l," +
+                            " rnd_bin(10, 20, 2) m," +
+                            " rnd_str(5,16,2) n," +
+                            " rnd_char() t," +
+                            " CAST(now() as LONG256) l256" + // Semi-random to not change saved txt file result
+                            " from long_sequence(500)" +
+                            ") timestamp (ts) partition by DAY",
+                    sqlExecutionContext);
+
+            // create a copy of 'x' as our expected result set
+            compiler.compile("create table y as (select * from x)", sqlExecutionContext);
+
+            // copy 'x' into itself, thus duplicating every row
+            compiler.compile("insert into x select * from x", sqlExecutionContext);
+
+            TestUtils.assertSqlCursors(
+                    compiler,
+                    sqlExecutionContext,
+                    "y",
+                    "select distinct * from x",
```

Comment:
How about 
select distinct * from x order by ts DESC

Reply:
this should work too, in this scenario `order by ts desc` will be discarded (table is already ordered) and distinct will work as expected, but i will add a test